src/build.d

```diff
@@ -1450,7 +1389,7 @@ class DependencyRef
 /**
 Takes a lambda and returns a memoized function to build a dependecy object.
 */
-alias makeDep(alias Func) = memoize!(function() {
+alias makeDep(alias Func, T...) = memoize!(function(T args) {
```

Comment:
I've tried different solutions to avoid explicit types separated from the identifiers but ran into different issues:

1. Explicit function: `undefined identifier R|T`
```suggestion
alias makeDep(R function(DepBuilder*, DependencyRef, T) Func, R, T...) = memoize!(function(T args) {
```
Rejected because it contains forward-references within the template parameter list
(This seems like an artificially restriction as e.g. template methods allow forward references to the return type but maybe I am underestimating the entailing implementation difficulties) 

2. Extracting parameter types using traits:
```suggestion
alias makeDep(alias Func) = memoize!(function(Parameters!Func[2..$] args) {
```
Fails as it can't resolve the parameter types (probably caused by the cyclic dependency between `Func` and this lambda. Could a resolution fur such a scenario be realized in DMD with reasonable effort?)

3. ?

Reply:
Nice. It would be beneficial to conflate the builder and the dependency as this requires all parameter types to be spelled out.

Or at least have `alias DependencyBuilder = MethodInitializer!DependencyRef*`

Reply:
meh, I'm fine with `MethodInitializer!DependencyRef*`.  It's a bit verbose but that doesn't bother me as it will only be used once.

Reply:
>  ... as it will only be used once
Currently, but we can keep the full type anyway

Reply:
Using `DependencyBuilder` adds a level of indirection.  Now you have to look up what a `DependencyBuilder` is and then see that it's a `MethodInitializer` and then look up what that is.  This can be worth it to save extra characters, but since it's only being used in one place, I don't see any benefit to it.

Reply:
Okay